[One Discover] Add summary column for logs contextual experience

tonyghiani opened the pull request:
## 📓 Summary

Closes https://github.com/elastic/logs-dev/issues/165

This work introduces a Summary column as a replacement for the Document one for the Discover logs contextual experience. 
> We also decided to port this change as a replacement for the **resource** and **content** virtual columns in Logs Explorer to have a better alignment between the 2 apps.

## 🎥 Demo

https://github.com/user-attachments/assets/3dfca483-768e-471c-8735-d5fc8bbd5d00

## 💡 Reviewer hints

I left notes through the changes to help answer some questions.

The notable changes on this PR are:
- Rename `Document` column to `Summary`.
- Refactor `resource` and `content` virtual columns into a single `Summary` column, which replaces the default Summary content for the logs' contextual experience.
- Provide support for the plugin services to the context awareness profiles.
- Disable virtual columns (clean up will go in a follow-up work) in Logs Explorer and rely on the Summary column as the default selection.


<!--ONMERGE {"backportTargets":["8.x"]} ONMERGE-->

Commit:
feat(discover): rename Document column to Summary

Commit:
feat(discover): support discover services in profiles definition

Commit:
refactor(discover): minor components adjustments

Commit:
refactor(discover): resource column

Commit:
feat(discover): responsive summary column

Commit:
feat(discover): custom popover

Commit:
feat(discover): custom popover mvp

Commit:
refactor(discover): update summary conditions

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!

Commit:
refactor(discover): rename virtual column service

Commit:
Merge branch 'feature/one-discover-summary-column' of github.com:tonyghiani/kibana into tonyghiani-feature/one-discover-summary-column-copy

Commit:
Merge branch 'tonyghiani-feature/one-discover-summary-column-copy' into feature/one-discover-summary-column

tonyghiani committed:
Merge branch 'main' into feature/one-discover-summary-column

Commit:
refactor(discover): minor cleanups

Commit:
refactor(discover): update popover content

tonyghiani committed:
Merge branch 'main' into feature/one-discover-summary-column

Commit:
refactor(discover): apply lazy loading on summary column

Commit:
refactor(discover): remove cell popover

Commit:
refactor(discover): remove unused file

Commit:
refactor(logs-explorer): switch to new summary column

Commit:
refactor(logs-explorer): disable resource and content as default columns

Commit:
Merge branch 'feature/one-discover-summary-column' of github.com:tonyghiani/kibana into tonyghiani-feature/one-discover-summary-column

Commit:
Merge branch 'tonyghiani-feature/one-discover-summary-column' into feature/one-discover-summary-column

Commit:
refactor(discover): move summary column files

Commit:
refactor(discover): align font size depending on grid params

Commit:
refactor(discover): fix broken tests

tonyghiani committed:
Merge branch 'main' into feature/one-discover-summary-column

Commit:
refactor(discover): fix i18n

tonyghiani commented:
/ci

tonyghiani committed:
Merge branch 'main' into feature/one-discover-summary-column

Commit:
test(discover): add basic testing for summary column

Commit:
Merge branch 'feature/one-discover-summary-column' of github.com:tonyghiani/kibana into feature/one-discover-summary-column

tonyghiani committed:
Merge branch 'main' into feature/one-discover-summary-column

tonyghiani commented:
/ci

Commit:
refactor(discover): update getFieldValue and remove duplicate getFieldFromDoc

tonyghiani commented on packages/kbn-discover-utils/src/utils/get_field_value.ts:
```diff
@@ -9,7 +9,10 @@
 import { DataTableRecord } from '../types';
 
-export const getFieldValue = (record: DataTableRecord, field: string) => {
+export const getFieldValue = <TRecord extends DataTableRecord, TField extends string>(
```
> [!NOTE]
> Make it generic to match and replace `getFieldFromDoc` util.

tonyghiani commented on packages/kbn-unified-data-table/src/utils/get_render_cell_popover.tsx:
```diff
@@ -28,6 +28,7 @@ export const getCustomCellPopoverRenderer = () => {
     useEffect(() => {
       setCellPopoverProps({
         panelClassName: 'unifiedDataTable__cellPopover',
+        panelProps: { css: { maxInlineSize: 'min(75vw, 600px) !important' } },
```
> [!NOTE]
> Force wider inline size on the popover panel to match the UI Design.

tonyghiani replied:
I updated this part to make it wider only for an allowed list of columns. 
Not the ideal fix as we open the door to visual inconsistency, but it should work for this scenario.
[fix(discover): make popover wide for limited columns](https://github.com/elastic/kibana/pull/192567/commits/e87d6a8400f6931dac16e64c60b39bd58fd6f8a0)

tonyghiani commented on src/plugins/discover/public/components/data_types/logs/cell_actions_popover.tsx:
```diff
@@ -0,0 +1,128 @@
+  }) => ReactElement;
+}
+
+export function CellActionsPopover({
```
> [!NOTE]
> `CellActionsPopover` is a refactor from the original `ChipWithPopover` component. 
> It now uses the Props Collection pattern to give control of the popover trigger to the `children`, and apply minor changes to align the generated HTML with the cell actions.

tonyghiani commented on src/plugins/discover/public/components/data_types/logs/log_level_badge_cell.tsx:
```diff
@@ -24,11 +23,12 @@ export const getLogLevelBadgeCell =
-        data-test-subj={dataTestSubj}
-        css={badgeCss}
-      />
+      <EuiFlexGroup alignItems="center" css={{ height: '100%' }}>
```
> [!NOTE]
> Full height & `alignItems` are more reliable than a fixed top-margin for centering the badge since we don't know how tall will be the cell given the different grid modes.

tonyghiani replied:
The issue here, as for badges in this UI, is that we are trying to squeeze something that doesn't have enough space in the grid for compact mode. The EuiBadge component has a predefined size, as the grid cells have fixed sizes to perform the virtual rendering, which makes it tricky to keep a proper alignment of the badges into less space than required.

The only way to properly align this content without relying on magic pixel numbers is to compute when we are in a single row or not, but that still doesn't completely fix alignment, since for custom mode with 1 row, the grid cell space is calculated differently and cut out whatever exceeds the available space. I'm not sure if there is a cheap workaround for this since we are heavily constrained by the DataGrid spacing.

davismcphee replied:
Unfortunately it looks like these changes never got reverted and are still present in the PR branch:
![image](https://github.com/user-attachments/assets/bca17cc0-ee7c-4114-9449-badefd9126ae)

Could we please revert these as the final task on this PR?

tonyghiani replied:
Ops my bad, I thought this was reverted already! Fixed, thanks again!

tonyghiani commented on src/plugins/discover/public/components/data_types/logs/summary_column/index.tsx:
```diff
@@ -7,11 +7,13 @@
-  const fieldValueArray = doc.flattened[field];
-  return fieldValueArray && fieldValueArray.length ? fieldValueArray[0] : undefined;
-};
+export const getSummaryColumn =
```
> [!IMPORTANT]
> This factory function creates the SummaryColumn component such that dependencies are injected as props and the logic for the column is loaded only when necessary (logs source, Logs Explorer, etc..)

tonyghiani commented on src/plugins/discover/public/components/data_types/logs/summary_column/resource.tsx:
```diff
@@ -0,0 +1,56 @@
+  alignItems?: EuiFlexGroupProps['alignItems'];
+}
+
+export const Resource = ({ row, limited = false, alignItems = 'stretch' }: ResourceProps) => {
```
> [!NOTE]
> - `Resource` ➡️ Interactive badges for filtering
> - `StaticResource` ➡️ Static badges for displaying only in the popover detail

tonyghiani commented on src/plugins/discover/public/components/data_types/logs/summary_column/summary_column.tsx:
```diff
@@ -0,0 +1,142 @@
+  rowHeight: number | undefined;
+}
+
+const SummaryColumn = (props: SummaryColumnProps & SummaryColumnFactoryDeps) => {
```
> [!NOTE]
> Act mainly as a factory component for the cell vs the popover detail.

tonyghiani commented on src/plugins/discover/public/components/data_types/logs/summary_column/utils.tsx:
> [!NOTE]
> All these utils are moved from the smaller files around the single components for the summary column.

tonyghiani commented on src/plugins/discover/public/context_awareness/profile_providers/common/logs_data_source_profile/accessors/get_cell_renderers.tsx:
```diff
@@ -7,17 +7,22 @@
-export const getCellRenderers: DataSourceProfileProvider['profile']['getCellRenderers'] =
-  (prev) => () => ({
-    ...prev(),
+export const makeGetCellRenderersHandler =
```
> [!NOTE]
> Convert into a factory function for the getter to inject `ProfileProviderServices`

Commit:
refactor(discover): minor code changes

tonyghiani commented:
/ci

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

elasticmachine commented:
Pinging @elastic/kibana-data-discovery (Team:DataDiscovery)

elasticmachine commented:
Pinging @elastic/obs-ux-logs-team (Team:obs-ux-logs)

Commit:
refactor(discover): update checks

Commit:
Merge branch 'feature/one-discover-summary-column' of github.com:tonyghiani/kibana into feature/one-discover-summary-column

Commit:
test(logs-explorer): skip tests for disabled features

tonyghiani committed:
Merge branch 'main' into feature/one-discover-summary-column

Commit:
refactor(discover): fix conflict

Commit:
refactor(discover): skip tests for disabled features

Commit:
refactor(discover): update test

tonyghiani committed:
Merge branch 'main' into feature/one-discover-summary-column

Commit:
Merge branch 'feature/one-discover-summary-column' of github.com:tonyghiani/kibana into tonyghiani-feature/one-discover-summary-column

Commit:
Merge branch 'tonyghiani-feature/one-discover-summary-column' into feature/one-discover-summary-column

Commit:
refactor(discover): update test

Kerry350 reviewed (approved):
Great work @tonyghiani 👏  Thanks as always for the inline comments.

I think I've tested everything in terms of the column itself, popover, actions, filtering, with various grid rendering options etc.

The partial cleanup of the virtual columns for now makes sense to me 👍

davismcphee commented on src/plugins/discover/public/plugin.tsx:
```diff
@@ -302,27 +302,41 @@ export class DiscoverPlugin
+    return { rootProfileService, dataSourceProfileService, documentProfileService };
+  }
+
+  private createProfilesManager = once(async (core: CoreStart, plugins: DiscoverStartPlugins) => {
```
These changes break context awareness in dashboards. Moving `once` to `createProfilesManager` makes the profiles manager a singleton, but each instance of Discover or the saved search embeddable should get its own profiles manager instance since it's tied to the application state. Otherwise the context awareness state of one panel gets mangled when another panel is added, like this:

https://github.com/user-attachments/assets/a7eb13e9-d54e-46e5-835d-7ae31f3217b3

The profile services were safe to be singletons since we only needed to register providers once on initialization and could share them between profiles manager instances, although I'm not sure if that's really the case now either since we're now passing dependencies to providers.

tonyghiani replied:
Good catch! I fixed it removing the memoization so that a new Profiles manager is created for each instance of the embeddable or usage in Discover.
I also agree that there is not such a performance impact in creating new profile services when required vs memorizing them, compared to the risk of undesired side effects as happened here.
I would go for not having this memo in place atm.

Here is the embeddable working as expected:

https://github.com/user-attachments/assets/9b4fd1dc-ffba-44de-9d6a-758f3f9aab7d

davismcphee reviewed (commented):
Finished my initial code review and testing, and it's looking great overall! I left a bit of feedback in the comments, but in general the implementation seems good. I did encounter a few things while testing to look into though (a lot are related to row height and density which are usually a pain to manage, especially with more complex layouts like this):

Vertical alignment is wonky for Summary cells with no badges when row height is auto fit or custom > 1:
![vertical_align_3](https://github.com/user-attachments/assets/34d745f3-ec28-4123-a1f2-75402b84e979)

It also seems to be off with row height = 1, but only slightly (maybe slight difference in font size?):
![vertical_align_1](https://github.com/user-attachments/assets/a6fcd350-6ab8-4bd3-b2f2-191208550fd7)

The cell content has an empty space at the start when row height = 1:
![horizontal_align](https://github.com/user-attachments/assets/2f735458-e4fa-4734-b6dc-63690cdec7b2)

The truncation ellipsis is not visible for custom row heights:
![ellipsis](https://github.com/user-attachments/assets/570d0b95-ab14-468f-b3b3-fbdda58a9397)

When row height = 2, the text content below the badges is not rendered:
![row_height_2](https://github.com/user-attachments/assets/fb863df8-5ef6-431b-9142-d75721ec7496)

When row height > 1, it looks like cells with `message` values display the text content on the same line as the badges, but ones without `message` values display the formatted result on a separate line. Is this intentional? It looks a bit odd at least IMO:
![message_same_line_custom](https://github.com/user-attachments/assets/31a5014f-189b-43e5-985f-bfce6d4f754e)

The Summary badge filtering doesn't work in ES|QL mode, and instead adds a DSL filter. We may be able to use the same filtering logic here that we currently use for filtering in non-Summary cells in ES|QL mode:
![esql_filter](https://github.com/user-attachments/assets/9d3be097-975f-43c8-8331-7082758d7cdb)

Badge filtering is inaccessible to keyboard users since it's disabled in the popover, and you can't enter expandable cells by keyboard in the data grid:

https://github.com/user-attachments/assets/2c6addd0-f9e9-4c2e-bb3b-bce1340dfca1

The text content below the badges is cut off for normal or expanded densities:
![density_cutoff](https://github.com/user-attachments/assets/a8940201-9a7a-41df-825a-dc1601bd870d)

The badges also don't respect the density setting, although this may be less of a concern:
![badge_density](https://github.com/user-attachments/assets/1fe4b9a9-663f-4387-a8a1-df9610b5ed72)

Some of the comments I left are performance related, mainly because we're fighting with bad performance when scrolling (especially for small row heights). Honestly it's bad for the default Summary renderer too, so in general any performance savings we can squeeze out of the renderers can be beneficial:

https://github.com/user-attachments/assets/c962b87a-6993-43fb-88e2-323f1b1c795e

Another thing is that some functional tests for these changes would be valuable here. I see a lot of the old Logs Explorer tests have been skipped, but there don't seem to be equivalent Summary column replacements for them. Maybe this was intended as part of the follow up to remove smart fields code? Personally I'd prefer to have some as part of the initial PR.

And lastly I need to confirm if we are good to merge this as part of the shared logs profile. Part of me feels like we discussed making it an O11y specific thing, but I honestly can't remember what the decision was...

tonyghiani commented on src/plugins/discover/public/components/data_types/logs/copy_button.tsx:
```diff
@@ -7,24 +7,23 @@
+      size="s"
+      iconType="copyClipboard"
+      aria-label={ariaCopyValueText}
+      onClick={() => copyToClipboard(value)}
```
TIL, thanks for the suggestion! I'll apply the change :)

Commit:
refactor(discover): use EuiCopy

Commit:
fix(discover): conditionally render resource block when missing for better alignment

Commit:
fix(discover): remove pre-optimization

Commit:
fix(discover): make popover wide for limited columns

tonyghiani commented on src/plugins/discover/public/components/data_types/logs/summary_column/content.tsx:
```diff
@@ -0,0 +1,85 @@
+
+const SourceDocument = dynamic(
+  () => import('@kbn/unified-data-table/src/components/source_document')
+);
```
Yep, this was a leftover from the initial virtual column implementation, we can remove it!
[fix(discover): remove pre-optimization](https://github.com/elastic/kibana/pull/192567/commits/e147b753eb05fdaf14915246f22f5c478728ecd1)

Commit:
fix(discover): pre-compute shouldShowFieldHandler once when creating the summary column

Commit:
fix(discover): types and test

Commit:
fix(discover): remove profile manager singleton

tonyghiani committed:
Merge branch 'main' into feature/one-discover-summary-column

Commit:
perf(discover): optimize document formatting to remove properties

tonyghiani commented on src/plugins/discover/public/components/data_types/logs/summary_column/utils.tsx:
```diff
@@ -0,0 +1,126 @@
+
+  // We need 2 loops here for flattened and raw.fields. Flattened contains all fields,
+  // whereas raw.fields only contains certain fields excluding _ignored
+  for (const fieldName in flattened) {
+    if (isFieldAllowed(fieldName) && flattened[fieldName]) {
+      flattenedResult[fieldName] = flattened[fieldName];
+    }
+  }
+
+  for (const fieldName in fields) {
+    if (isFieldAllowed(fieldName) && fields[fieldName]) {
+      rawFieldResult[fieldName] = fields[fieldName];
+    }
+  }
```
I agree 100% with your concern here, it might become a heavy task to process when it has many fields.
This is how it used to work with virtual columns too, so I guess we would have the same performance issue there.
As an initial improvement, I refactored its usage in the Content sub-component to only format the document when we need it, so in case none of the defined fields for the content exists and we have to fallback to the whole document.

This should at least mitigate the number of documents we have to process this way and, in an ideal scenario, reduce the need for it to zero.

[perf(discover): optimize document formatting to remove properties](https://github.com/elastic/kibana/pull/192567/commits/933b08013c589ab0c6ba4e9f61257daff24446ac)

P.S. I also searched for any possible improvement at the low level code, bug given we have to assert each property against a prefix and not a the full name, it seems there is not much margin for improvement.

Commit:
fix(discover): remove import

tonyghiani commented:
@davismcphee we are having an offline conversation with @ryankeairns about the design. Many of the alignment/spacing issue you (correcly) pointed out are due to the fact the badges don't fit well into the DataGrid in compact mode given the available space. We'll continue talking about this, just warning there might be design changes.

cc. @gbamparop @ninoslavmiskovic @flash1293 @LucaWintergerst

ryankeairns commented:
@tonyghiani The next EUI release will contain updates to data grid (https://github.com/elastic/eui/pull/8025) that include a fix for cell height. This should address the current variation in height between Single and Custom = 1 settings (i.e. the badge getting clipped).

Assuming that fixes the original issue, I will hold off on the alternative, non-badge design idea. Let me know if you would still like to see that for any other reason(s).

tonyghiani committed:
Merge branch 'main' into feature/one-discover-summary-column

Commit:
refactor(discover): update styles

Commit:
Merge branch 'feature/one-discover-summary-column' of github.com:tonyghiani/kibana into feature/one-discover-summary-column

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

Commit:
refactor(discover): update styles

Commit:
Merge branch 'feature/one-discover-summary-column' of github.com:tonyghiani/kibana into feature/one-discover-summary-column

Commit:
feat(discover): apply shared actions to summary column

Commit:
refactor(discover): update test

tonyghiani commented on src/plugins/discover/public/application/context/context_app_content.tsx:
```diff
@@ -173,7 +178,10 @@ export function ContextAppContent({
     const getCellRenderers = getCellRenderersAccessor(() => ({}));
-    return getCellRenderers();
+    return getCellRenderers({
+      density: DataGridDensity.COMPACT,
+      rowHeight: ROWS_HEIGHT_OPTIONS.single,
```
I fixed the alignment and looks better, but I see the issue here.

TBH, purely using local storage stored values for this page grid configuration seems unreliable, as landing on the page with a clean cache would also result in a different view than what is configured on the Discover Grid.

Also, this is not a short refactor to take over on this PR, so I don't think we have a quick workaround, and directly accessing the local storage values seems less stable as it relies on a feature implemented by the child UnifiedDataTable.

Do you have any recommendations on how you would take over this? You surely have wider knowledge of this part of the codebase than me 😅

tonyghiani replied:
Thanks for the PR, just merged and works well!

Commit:
test(discover): add FT for summary column displaying

tonyghiani commented:
Hi @davismcphee, sorry for the late response, I've addressed most of your suggestions and spent time investigating the design issues you reported.

> - [x]  Vertical alignment is wonky for Summary cells with no badges when row height is auto fit or custom > 1
> - [x]  It also seems to be off with row height = 1, but only slightly (maybe slight difference in font size?)
> - [x] The cell content has an empty space at the start when row height = 1:
> - [ ] The truncation ellipsis is not visible for custom row heights:
> - [x] When row height = 2, the text content below the badges is not rendered:
> - [x] The text content below the badges is cut off for normal or expanded densities:

These issues were mostly caused by the text size, conditional rendering and the badge excessive size, I've been discussing workarounds/solutions with Ryan and the latest EUI changes should allow us to fix most of them.

> When row height > 1, it looks like cells with message values display the text content on the same line as the badges, but ones without message values display the formatted result on a separate line. Is this intentional? It looks a bit odd at least.

This doesn't depend on the `message` existence, it was about the content length. 
If the message fits in a single line, it could go next to the badge.
It was initially intentional, but not necessarily a final version. I switched to always have the content on a second line when row height > 1👌 

> The Summary badge filtering doesn't work in ES|QL mode, and instead adds a DSL filter. We may be able to use the same filtering logic here that we currently use for filtering in non-Summary cells in ES|QL mode.

I fixed it with [feat(discover): apply shared actions to summary column](https://github.com/elastic/kibana/pull/192567/commits/42a4501585afba4a09614c37df901254477c136b), although I'm not particularly happy because the implementation promotes prop drilling at many levels, which increases the maintainability effort for this code. I spent some time refactoring, but it was widely affecting the scope of this work. It might be worth discussing this in future and tackling some refactoring to reduce the props drilling of these data table actions.

> Badge filtering is inaccessible to keyboard users since it's disabled in the popover, and you can't enter expandable cells by keyboard in the data grid.

As per [this comment on Figma](https://www.figma.com/design/CpEvYxYccT8d2ZrxuhQakc?node-id=237-41882#885230390), this was a design choice, I see it comes with an accessibility issue. 
Usually, the pattern is that the cell content is not interactive, and once the popover is opened, the focus goes there and the user can navigate the content with the keyboard.
@ryankeairns should we change it so that also the badges in the popover are not purely presentational, but have the same popup for filtering that is shown when clicking the cell badges? I don't see another way otherwise, as the cell content is not interactive by default (I tried interactive wrappers or forcing tab index on those badges, but the `EuiDataGrid` component is quite strict on this and changes everything back to not interactive.)

> The badges also don't respect the density setting, although this may be less of a concern:

I think you mean the size of the badges doesn't scale with density changes. If so, `EuiBadge` has a unique size, so we cannot make it bigger/smaller based on the grid density.

> Another thing is that some functional tests for these changes would be valuable here. I see a lot of the old Logs Explorer tests have been skipped, but there don't seem to be equivalent Summary column replacements for them. Maybe this was intended as part of the follow up to remove smart fields code? Personally I'd prefer to have some as part of the initial PR.

The skipped tests are purely focused to check if the virtual columns were rendered. For the summary column, I added a Jest test for the content, and with [test(discover): add FT for summary column displaying](https://github.com/elastic/kibana/pull/192567/commits/ed9bc98aba5f221a973737ec9248138d90c6264e) I updated the FT test for the cell renderer extension point, I think is ok to start like this since we might change more on this column soon.

> And lastly I need to confirm if we are good to merge this as part of the shared logs profile. Part of me feels like we discussed making it an O11y specific thing, but I honestly can't remember what the decision was...

Af far as I remember, this was intended to be a logs feature, as it purely relies on logs concepts, there is no specific Observability knowledge in the rendered data (apart from navigation to entities).

Commit:
test(discover): update FT test

ryankeairns commented:
@l-suarez I don't know what your workload looks like but, if you have some bandwidth, this one could use design input.

> Badge filtering is inaccessible to keyboard users since it's disabled in the popover, and you can't enter expandable cells by keyboard in the data grid.

> > As per [this comment on Figma](https://www.figma.com/design/CpEvYxYccT8d2ZrxuhQakc?node-id=237-41882#885230390), this was a design choice, I see it comes with an accessibility issue.
Usually, the pattern is that the cell content is not interactive, and once the popover is opened, the focus goes there and the user can navigate the content with the keyboard.
@ryankeairns should we change it so that also the badges in the popover are not purely presentational, but have the same popup for filtering that is shown when clicking the cell badges? I don't see another way otherwise, as the cell content is not interactive by default (I tried interactive wrappers or forcing tab index on those badges, but the EuiDataGrid component is quite strict on this and changes everything back to not interactive.)

@tonyghiani I recall that thread and tried this on the oblt instance where this capability still exists. Using it makes me now feel it is better to leave it as it was (keyboard accessible; popovers on top of a modal) rather than remove the functionality. I don't have bandwidth to take this on - and I don't know that @l-suarez does either - so, in the meantime, I believe it would be preferable to keep it keyboard accessible until a more-complete alternative arises.

tonyghiani committed:
Merge branch 'main' into feature/one-discover-summary-column

Commit:
fix(discover): conflicts

logeekal reviewed (approved):
Looks good from the investigations side. Thanks.

Commit:
refactor(discover): update styles and cleanup

Commit:
refactor(discover): add popover close button

Commit:
refactor(discover): remove unnecessary wrapper

davismcphee committed:
Fix default grid density and row height passed to getCellRenderers

davismcphee reviewed (commented):
Great work on the updates! Basically all of the issues I reported have been resolved now, and it's looking nice and polished 👌 There are just a couple of final things left to address and this will be good to go on my end. We'll also clarify tomorrow in the sync that we're good to merge this into the generic logs profile just to be certain.

>I think you mean the size of the badges doesn't scale with density changes.

Yeah I was referring to the text size of the badges, but if the scale is hardcoded, we're definitely good to leave it.

>I updated the FT test for the cell renderer extension point, I think is ok to start like this since we might change more on this column soon.

Agreed, these tests should be good for now, thanks!

tonyghiani commented on src/plugins/discover/public/components/data_types/logs/summary_column/content.tsx:
```diff
@@ -7,79 +7,116 @@
+      <EuiDescriptionList
+        type="inline"
+        compressed={isCompressed}
+        className="unifiedDataTable__descriptionList unifiedDataTable__cellValue"
```
Agree, the simple fix for a clean HTML is actually to just let inherit the styles from the parent, I fixed it!

Commit:
refactor(discover): change requests

tonyghiani committed:
Merge pull request #2 from davismcphee/fix-default-density-and-row-height

Fix default grid density and row height passed to `getCellRenderers`

Commit:
refactor(discover): fix imports

Commit:
refactor(discover): remove unused property

davismcphee reviewed (commented):
All of the remaining comments have been resolved, and we got confirmation we're good to merge this into the generic logs profile. There's just one _tiny_ thing left to do (revert some changes), then I promise I'll get out of the way and let this get merged 🙏

Commit:
refactor(discover): revert log level badge styles

tonyghiani committed:
Merge branch 'main' into feature/one-discover-summary-column

davismcphee reviewed (approved):
Looks like we're all good to go now, so let's merge this thing 🤘 Awesome work on this, thanks! It's a huge step forward for One Discover and I'm excited to get it out to users soon.

tonyghiani committed:
Merge branch 'main' into feature/one-discover-summary-column

kibanamachine commented:
Starting backport for target branches: 8.x

https://github.com/elastic/kibana/actions/runs/11102974432

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/194389">](https://github.com/elastic/kibana/pull/194389)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 9.4.3 -->
